Samples/MapScenarios/MapScenarios/Scenarios/WalkTime/Views/WalkTimeView.swift

```diff
@@ -0,0 +1,98 @@
+//
+//  WalkTimeView.swift
+//  MapScenarios
+//
+//  Created on 2/21/19.
+//  Copyright © 2019 Patrick Dunshee. All rights reserved.
+//
+
+import UIKit
+import CoreLocation
+
+extension Notification.Name {
+    static let ExitWalkTimeButtonTapped = NSNotification.Name("ExitWalkTimeButtonTapped")
+}
+
+extension Collection where Element: Numeric {
+    // Returns the total sum of all elements in the array
+    var total: Element { return reduce(0, +) }
+}
+
+extension Collection where Element: BinaryInteger {
+    // Returns the average of all elements in the array
+    var average: Double {
+        return isEmpty ? 0 : Double(total) / Double(count)
+    }
+}
+
+extension Collection where Element: BinaryFloatingPoint {
+    // Returns the average of all elements in the array
+    var average: Element {
+        return isEmpty ? 0 : total / Element(count)
+    }
+}
+
+class WalkTimeView: UIView {
```

Comment:
Can we separate the calculations from the view? maybe into a manager class or something, but i could see a developer wanting our time remaining calculations without being coupled to the UI

Reply:
That might be true, but I'd like put the code in one place(less places) as possible as we can, then developer has less copy and paste to integrate the feature.

Reply:
Converting to MVP might help with this to insulate the logic from the view. See comment above.